hist/hist/src/TGraph.cxx

```diff
@@ -2008,6 +2011,70 @@ Int_t TGraph::RemovePoint(Int_t ipoint)
    return ipoint;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Save the graph as .csv, .tsv or .txt. In case of any other extension, fall
+/// back to TObject::SaveAs
+///
+/// The result can be immediately imported into Excel, gnuplot, Python or whatever,
+/// without the needing to install pyroot, etc.
+///
+/// \param filename the name of the file where to store the graph
+/// \param option some tuning options
+///
+/// The file extension defines the delimiter used:
+///  - `.csv` : comma
+///  - `.tsv` : tab
+///  - `.txt` : space
+///
+/// If option = "title" a title line is generated with the axis titles.
+
+void TGraph::SaveAs(const char *filename, Option_t *option) const
+{
+   TString del = "";
+   TString ext = "";
+
+   if (filename) {
+      if      (strstr(filename,".csv")) {del = ",";  ext = "csv";}
+      else if (strstr(filename,".tsv")) {del = "\t"; ext = "tsv";}
+      else if (strstr(filename,".txt")) {del = " ";  ext = "txt";}
+   }
+   if (del.Length()>0) {
+      std::ofstream out;
+      out.open(filename, std::ios::out);
+      if (!out.good ()) {
+         Error("SaveAs", "cannot open file: %s", filename);
+         return;
+      }
+      if (this->InheritsFrom(TGraphErrors::Class()) ) {
+         if(strstr(option,"title"))
+         out << "# " << GetXaxis()->GetTitle() << "\tex\t" << GetYaxis()->GetTitle() << "\tey" << std::endl;
+         double *ex = this->GetEX();
+         double *ey = this->GetEY();
+         for(int i=0 ; i<fNpoints ; i++)
+         out << fX[i] << del.Data() << (ex?ex[i]:0) << del.Data() << fY[i] << del.Data() << (ey?ey[i]:0) << std::endl;
+      } else if (this->InheritsFrom(TGraphAsymmErrors::Class()) || this->InheritsFrom(TGraphBentErrors::Class())) {
+         if(strstr(option,"title"))
+         out << "# " << GetXaxis()->GetTitle() << "\texl\t" << "\texh\t" << GetYaxis()->GetTitle() << "\teyl" << "\teyh" << std::endl;
+         double *exl = this->GetEXlow();
+         double *exh = this->GetEXhigh();
+         double *eyl = this->GetEYlow();
+         double *eyh = this->GetEYhigh();
+         for(int i=0 ; i<fNpoints ; i++)
+         out << fX[i] << del.Data() << (exl?exl[i]:0) << del.Data() << (exh?exh[i]:0) << del.Data() << fY[i] << del.Data() << (eyl?eyl[i]:0) << del.Data() << (exh?eyh[i]:0) << std::endl;
```

Comment:
```suggestion
         out << fX[i] << del.Data() << (exl?exl[i]:0) << del.Data() << (exh?exh[i]:0) << del.Data() << fY[i] << del.Data() << (eyl?eyl[i]:0) << del.Data() << (eyh?eyh[i]:0) << std::endl;
```

Reply:
Thanks for this! only the `eyh` must be fixed. The other comments are more subjective ;).